lib/symboldatabase.cpp

```diff
@@ -5724,10 +5724,9 @@ const Type* SymbolDatabase::findType(const Token *startTok, const Scope *startSc
             }
         } else {
             const Type * type = scope->findType(tok->str());
-            const Scope *scope1;
             if (type)
                 return type;
-            else if ((scope1 = scope->findRecordInBase(tok->str()))) {
+            else if (const Scope *scope1 = scope->findRecordInBase(tok->str())) {
```

Comment:
This is probably something which would fall under "scope reduction". I will file a ticket about this.

Reply:
https://trac.cppcheck.net/ticket/11221